docs/design/vertical_fallback.md

```diff
@@ -0,0 +1,163 @@
+
+**Cons:**
+
+- Larger index arrays, which *may* degrade performance for some locales
+
+There are two ways to resolve the locales, and they both require additional pre-processing on the locale before passing it into the data provider: *minimizing* or *maximizing*.
+
+#### Build-Time Option 1: Minimized Locales
+
+With this option, we assume that locales passed into the data provider are in minimal form. For example, `sr-Latn-ME` should be `sr-ME` and `de-Latn-LI` should be `de-LI`. This means that *every single data key* in `sr-ME` needs to have a direct pointer to the Latin version, but `de-LI` can directly fall back to `de` for most keys.
+
+**Pros:**
+
+- The industry tends to use minimized locales by default (`zh-TW`, `sr-ME`, …), so this option requires less pre-processing on locales at runtime.
+
+**Cons:**
+
+- Footgun: If a non-minimized locale is passed at runtime, it could have the incorrect behavior. For example, `sr-Latn-ME` or `de-Latn-LI` may not find the correct data.
+- Struggles to scale when there are many data keys
+
+#### Build-Time Option 2: Manually Maximized Locales
+
+With this option, we assume that locales are in maximal form. For example, `sr-ME` should be `sr-Latn-ME`, and `de-LI` should be `de-Latn-LI`. This means that every locale always has the correct script in its fallback chain. (The default script will continue to be pass-through.)
+
+This solves the problem with Option 1 where locales having a non-default script need to override every single data key. However, locales having a meta-region fallback will still need to carry all relevant overrides. But, I am considering this a much smaller problem than the script overrides, because (1) the data is in the correct script, so only a subset of keys need overrides, and (2) the number of locales having a meta-region in their fallback chain is very limited.
+
+Since maximized locales always have at least three subtags (language, script, and region), we can (and should) check in the data provider whether the input locale is maximized, and if it is not, we can return an error.
+
+**Pros:**
+
+- Scales well to many data keys
+
+**Cons:**
+
+- Not ergonomic; most users will need to call `.maximize()` before giving their locale to ICU4X
+
+#### Build-Time Option 3: Automatically Maximized Locales
+
+This is the same as option 2, except that instead of returning an error if the input locale is not maximized, we automatically maximize it, using the `LocaleCanonicalizer`.
+
+It is likely that if we shipped this option, we would give ICU4X users the ability to turn off the auto-maximization if they prefer.
+
+**Pros:**
+
+- Scales well to many data keys
+- Shares code and data with an existing component, rather than introducing a novel chunk of code and data as in the Runtime Approach
+
+**Cons:**
+
+- Requires shipping the likely subtags code and data
+- All lookups incur a small but nonempty penalty when maximization occurs
```

Comment:
I said my piece in the ICU-TC meeting the other day, but I think I vote for Option 3, with a provision to fall back to Option 2 for implementations that know their locale data doesn't need it and don't want to incur the space penalty for the likely-subtags tables.